ui/src/plugins/borealis/components/BorealisTypography.tsx

```diff
@@ -1,12 +1,21 @@
 import { Typography } from '@mui/material';
 import { EnrichedTypography, useBorealisEnrichSelector, type EnrichedTypographyProps } from 'borealis-ui';
+import { ApiConfigContext } from 'components/app/providers/ApiConfigProvider';
 import type { PluginTypographyProps } from 'components/elements/PluginTypography';
-import { memo, type FC } from 'react';
+import { memo, useContext, useState, type FC } from 'react';
 
 const BorealisTypography: FC<PluginTypographyProps> = ({ children, value, context, ...props }) => {
   const guessType = useBorealisEnrichSelector(ctx => ctx.guessType);
 
-  const type = guessType(value);
+  const [type, setType] = useState<string>('');
+
+  const { config } = useContext(ApiConfigContext);
+
+  if (config.configuration?.mapping[value]) {
+    setType(config.configuration?.mapping[value]);
+  } else {
+    setType(guessType(value));
+  }
```

Comment:
You should never run a setState function in the mainline of a component, as the state will be set on every render. big time no no. It's also not needed here, as neither function call is an expensive one:

```suggestion
  const type = config.configuration?.mapping[value] ?? guessType(value);
```